When building an image with USE_DEPMOD="0" we expect that on the first boot the
modutils.sh initscript to create the modules.dep file if it is empty or not
present.

Upstream-Status: Pending
Signed-off-by: Diego Sueiro <diego.sueiro@arm.com>

diff --git a/modutils.sh b/modutils.sh
index a78adf5729..4afd12a9fe 100755
--- a/modutils.sh
+++ b/modutils.sh
@@ -13,14 +13,17 @@
 
 LOAD_MODULE=modprobe
 [ -f /proc/modules ] || exit 0
-[ -f /etc/modules ] || [ -d /etc/modules-load.d ] || exit 0
-[ -e /sbin/modprobe ] || LOAD_MODULE=insmod
 
-if [ ! -f /lib/modules/`uname -r`/modules.dep ]; then
+# Test if modules.dep exists and has a size greater than zero
+if [ ! -s /lib/modules/`uname -r`/modules.dep ]; then
 	[ "$VERBOSE" != no ] && echo "Calculating module dependencies ..."
-	depmod -Ae
+	depmod -ae
 fi
 
+[ -f /etc/modules ] || [ -d /etc/modules-load.d ] || exit 0
+[ -e /sbin/modprobe ] || LOAD_MODULE=insmod
+
+
 loaded_modules=" "
 
 process_file() {
